app/src/main/java/com/example/parkingsystem/activity/ReservationActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.example.parkingsystem.activity;
+
+import android.app.DatePickerDialog;
+import android.app.TimePickerDialog;
+import android.os.Bundle;
+import android.widget.DatePicker;
+import android.widget.TimePicker;
+import androidx.appcompat.app.AppCompatActivity;
+import com.example.parkingsystem.databinding.ActivityReservationBinding;
+import com.example.parkingsystem.mvp.contract.ReservationContract;
+import com.example.parkingsystem.mvp.model.ReservationModel;
+import com.example.parkingsystem.mvp.presenter.ReservationPresenter;
+import com.example.parkingsystem.mvp.view.ReservationView;
+
+public class ReservationActivity extends AppCompatActivity implements DatePickerDialog.OnDateSetListener, TimePickerDialog.OnTimeSetListener {
+
+    private ActivityReservationBinding binding;
+    private ReservationContract.ReservationPresenterContract presenter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityReservationBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new ReservationPresenter(new ReservationModel(), new ReservationView(this, binding));
+
+        binding.buttonReservationActivityStartDate.setOnClickListener(view -> presenter.createDate(this));
+        binding.buttonReservationActivityFinishDate.setOnClickListener(view -> presenter.createDate(this));
+        binding.buttonReservationActivitySave.setOnClickListener(view -> finish());
```

Comment:
same here, the activity does not do the finish().

Reply:
Fixed